Prolific/Prolific/Utils/Persistence/DAO.m

```diff
@@ -325,6 +328,39 @@ - (void)getProjectWithId:(NSString *)projectId
     }];
 }
 
+#pragma mark - Cloud Storage
+
+- (void)uploadProfileImage:(NSData *)imageData
+                   forUser:(User *)user
+                completion:(void(^)(NSURL *downloadURL, NSError *error))completion {
+    NSString *const currUserId = user.userId;
+    
+    FIRStorageReference *const storageRef = [_storage reference];
+    FIRStorageReference *const profileImagesRef = [storageRef child:kProfileImagesRef];
+    FIRStorageReference *const profileImageRef = [profileImagesRef child:[NSString stringWithFormat:@"%@.png", currUserId]];
+    
+    FIRStorageMetadata *const uploadMetadata = [[FIRStorageMetadata alloc] init];
+    uploadMetadata.contentType = @"image/png";
+    
+    FIRStorageUploadTask *const uploadTask = [profileImageRef putData:imageData
+                                                             metadata:uploadMetadata
+                                                           completion:^(FIRStorageMetadata *metadata,
+                                                                        NSError *error) {
+        if (error != nil) {
+            completion(nil, error);
+        } else {
+            [profileImageRef downloadURLWithCompletion:^(NSURL * _Nullable URL, NSError * _Nullable error) {
+                if (error != nil) {
+                    completion(nil, error);
+                } else {
+                    NSURL *downloadURL = URL;
```

Comment:
seems like you don't need this pointer. You can just pass URL into the completion